plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/auth/basic/PinotBasicAuthenticationProvider.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.pinot.auth.basic;
+
+import io.trino.plugin.pinot.auth.PinotAuthenticationProvider;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
+public class PinotBasicAuthenticationProvider
+        implements PinotAuthenticationProvider
+{
+    private final Optional<String> authToken;
+
+    public PinotBasicAuthenticationProvider(String user, String password)
+    {
+        requireNonNull(user, "user is null");
+        requireNonNull(password, "password is null");
+        this.authToken = Optional.of(encode(user, password));
+    }
+
+    @Override
+    public Optional<String> getAuthenticationToken()
+    {
+        return authToken;
+    }
+
+    private String encode(String username, String password)
+    {
+        String auth = username + ":" + password;
+        byte[] encodedAuth = Base64.getEncoder().encode(auth.getBytes(StandardCharsets.ISO_8859_1));
+        return "Basic " + new String(encodedAuth, StandardCharsets.UTF_8);
+    }
+}
```

Comment:
Please use library to build credential. For instance, Okhttp has `Credentials.basic`.